### PR TITLE
Minor Improvement in atoms() and Tests

### LINKED ISSUE #121: expr.atoms() seems to be broken on 0.1.0.dev
In [19]: import symengine

	In [20]: import sympy

	In [21]: (symengine.var('x') + symengine.var('y')).atoms()
	Out[21]: set()

	In [22]: (sympy.var('x') + sympy.var('y')).atoms()
	Out[22]: {x, y}

on symengine.py=='0.1.0.dev' and symengine==0.2.0  (both installed using conda). On #59  `atoms` has been checked off as done. `free_symbols` seems to work fine.

	In [25]: (sympy.var('x') + sympy.var('y')).free_symbols
	Out[25]: {x, y}

	In [26]: (symengine.var('x') + symengine.var('y')).free_symbols
	Out[26]: {x, y}

### PR DESCRIPTION
**Pull request by ShikharJ**
Fixes #121 
@isuruf A similar function is implemented for DenseMatrix also. Are the changes required there too?

### CODE REVIEW

File: symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -588,8 +588,12 @@ cdef class Basic(object):
         s = set()
         if (isinstance(self, types)):
             s.add(self)
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx]: You can add the above 3 lines inside `if types` clause.

**Comment by ShikharJ** [comment]
Updated @isuruf.

File: symengine/lib/symengine_wrapper.pyx [hunk 2]
```diff
@@ -586,10 +586,14 @@ cdef class Basic(object):
...
-        for arg in self.args:
-            s.update(arg.atoms(*types))
+        if types:
+            if (isinstance(self, types)):
+                s.add(self)
+            for arg in self.args:
+                s.update(arg.atoms(*types))
+        else:
+            for arg in self.args:
+                s.update(arg.free_symbols)
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx]: You can just return `arg.free_symbols`

**Comment by ShikharJ** [comment]
Updated again @isuruf. Can you please take a final look?

**Pull request merged**